src/web/client/utilities/fileAndEntityUtil.ts

```diff
@@ -29,6 +29,17 @@ export function getFileEntityName(fileFsPath: string) {
         ?.entityName as string ?? WebExtensionContext.getVscodeWorkspaceState(fileFsPath)?.entityName as string;
 }
 
+export function getFileRootWebPageId(fileFsPath: string) {
+    const entityId = getFileEntityId(fileFsPath);
+    return (
+        (WebExtensionContext.entityDataMap.getEntityMap.get(entityId)
```

Comment:
Do add unit tests to make sure this logic is robust

Reply:
sure! will add in a new PR